### ISSUE TITLE
Hadolint should warn about errexit when using heredoc style RUN commands

### ISSUE THREAD

**Issue by twelvelabs** [hadolint/hadolint#882]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Hadolint warns me about setting `errexit` in the shell when writing [heredoc style RUN commands](https://www.docker.com/blog/introduction-to-heredocs-in-dockerfiles/). It should suggest something like:

```Dockerfile
SHELL ["/bin/bash", "-o", "errexit", "-c"]
# or
SHELL ["/bin/bash", "-o", "errexit", "-o", "pipefail", "-c"]
```

### Actual behavior

No warning is shown.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.10.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.4

RUN <<EOF
    something-that-returned-non-zero
    # These two were still executed
    something-else1
    something-else2
    # The RUN command completed successfully
EOF
```

I understand _why_ the above happens, but it's still a little surprising if you're used to the `RUN foo && bar` syntax and not having to guard against this. Luckily the command that failed did so noisily and I was able to catch it. If it had been more subtle it could have taken a while to track down. Seems like a good candidate for a linting rule.
